rts/Game/GameHelper.cpp

```diff
@@ -706,10 +709,16 @@ size_t CGameHelper::GenerateWeaponTargets(const CWeapon* weapon, const CUnit* av
 				if (sqDist2D > Square(modRange))
 					continue;
 
+				const float3 worldTargetDir = (targetPos - ownerPos).SafeNormalize();
+				const float angleOffset =  (1.f - worldMainDir.dot(worldTargetDir));
+				const float angleMod = angleOffset * weaponAimAdjustPriority + 1.f;
+				const float angleMul = angleMod*angleMod;
```

Comment:
Is this correct? Why is the multiplier the modifier squared?

Reply:
That's intentional. A linear distribution was placing insufficient emphasis on the importance of focusing on what's infront of the unit relative to say at 90deg or a 180deg.

Reply:
In that case
```suggestion
				const float angleMul = angleMod*angleMod; // linear would place insufficient [fixme, what does that mean? who decides what is important] importance on frontal compared to say 90'
```

Reply:
That is a very good point. I really should explain all this in more detail. Thanks for reminding me!

Reply:
Ideally the angle (and the other modifiers taken into account for the default engine priority - target health, distance etc) would also be reported to the `AllowWeaponTarget` call-in so that the game can decide how best to make use of that value. A game can already use various `Spring.GetUnitBla` functions to get those, including the angle, but it's prohibitively expensive.

Reply:
The plan is to seek feedback on the default angle aiming sensitivity, and make adjustments to improve the results. The current value is what came up as a reasonable starting point from some basic testing. I'm hoping this can be refined.

Reply:
> Would ZK use that feature?

Which one?
 * I'd expect ZK is most likely going to use fast reaim where applicable, though we don't actually have that many units with rapid-fire, high-velocity weapons powerful enough to kill more than 2+ light units per second so the effect is going to be subtle at any rate.
 * ZK currently overrides priority for ~all targeting because the default priority weights used to be, and might still be, inaligned with ZK assumptions. So it isn't going to use the angle prioritisation part as it exists now.
 * if angle was supplied to `AllowWeaponTarget` it would most likely see use eventually, though it's hard to tell. ZK has more extreme units than BAR and less blobs of identical units so it often makes more sense to target a juicy thing at a bad angle than to shoot a brick in front of you, so it would either take some time to tweak the weighting of the value to make sense or be relegated to be essentially a tiebreaker for equivalent targets. We're perenially low on mana so if you're just asking to decide whether implement that or not then I think I'd prefer if this wasn't prioritized.

@GoogleFrog ping if you want to add something.

Reply:
I was thinking about your suggestion: "Ideally the angle (and the other modifiers taken into account for the default engine priority - target health, distance etc) would also be reported to the AllowWeaponTarget". Thanks for your thoughts on that. Sorry to hear about the low-mana, I hope it isn't due to anything too serious.

Reply:
Adding an angle weight which maxes out at around 0.1 sounds like a decent addition to ZK targeting.